src/energyplus/Test/ZoneHVACWaterToAirHeatPump_GTest.cpp

```diff
@@ -0,0 +1,161 @@
+#include "../../model/ThermalZone.hpp"
+#include "../../model/Space.hpp"
+#include "../../model/ZoneHVACWaterToAirHeatPump.hpp"
+#include "../../model/CoilHeatingWaterToAirHeatPumpEquationFit.hpp"
+#include "../../model/CoilCoolingWaterToAirHeatPumpEquationFit.hpp"
+#include "../../model/FanOnOff.hpp"
+#include "../../model/CoilHeatingElectric.hpp"
+#include "../../model/ScheduleConstant.hpp"
+
+#include <utilities/idd/ZoneHVAC_WaterToAirHeatPump_FieldEnums.hxx>
+#include <utilities/idd/Coil_Heating_WaterToAirHeatPump_EquationFit_FieldEnums.hxx>
+#include <utilities/idd/Coil_Cooling_WaterToAirHeatPump_EquationFit_FieldEnums.hxx>
+
+#include <utilities/idd/IddEnums.hxx>
+#include "../../utilities/idf/IdfObject.hpp"
+#include "../../utilities/idf/IdfObject_Impl.hpp"
+
+#include "../../utilities/idf/WorkspaceObject.hpp"
+#include "../../utilities/idf/WorkspaceObject_Impl.hpp"
+
+using namespace openstudio::energyplus;
+using namespace openstudio::model;
+using namespace openstudio;
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_ZoneHVACWaterToAirHeatPump) {
+
+  Model m;
+
+  Schedule sch = m.alwaysOnDiscreteSchedule();
+  CoilHeatingWaterToAirHeatPumpEquationFit htg_coil(m);
+  CoilCoolingWaterToAirHeatPumpEquationFit clg_coil(m);
+  FanOnOff fan(m);
+  CoilHeatingElectric supp_htg_coil(m);
+
+  ZoneHVACWaterToAirHeatPump hp(m, sch, fan, htg_coil, clg_coil, supp_htg_coil);
+
+  // Need to be in a thermal zone to be translated, and TZ needs at least one space
+  ThermalZone z(m);
+  hp.addToThermalZone(z);
+  Space s(m);
+  s.setThermalZone(z);
+
+  openstudio::energyplus::ForwardTranslator ft;
+  Workspace w = ft.translateModel(m);
+
+  WorkspaceObjectVector idf_hps(w.getObjectsByType(IddObjectType::ZoneHVAC_WaterToAirHeatPump));
+  EXPECT_EQ(1u, idf_hps.size());
+  WorkspaceObject idf_hp(idf_hps[0]);
+
+  WorkspaceObjectVector idf_ccs(w.getObjectsByType(IddObjectType::Coil_Cooling_WaterToAirHeatPump_EquationFit));
+  EXPECT_EQ(1u, idf_ccs.size());
+  WorkspaceObject idf_cc(idf_ccs[0]);
+
+  WorkspaceObjectVector idf_hcs(w.getObjectsByType(IddObjectType::Coil_Heating_WaterToAirHeatPump_EquationFit));
+  EXPECT_EQ(1u, idf_hcs.size());
+  WorkspaceObject idf_hc(idf_hcs[0]);
+
+  EXPECT_EQ(hp.nameString(), idf_hp.getString(ZoneHVAC_WaterToAirHeatPumpFields::Name).get());
+  EXPECT_EQ(sch.nameString(), idf_hp.getString(ZoneHVAC_WaterToAirHeatPumpFields::AvailabilityScheduleName).get());
+  EXPECT_NE("", idf_hp.getString(ZoneHVAC_WaterToAirHeatPumpFields::AirInletNodeName, false).get());
+  EXPECT_NE("", idf_hp.getString(ZoneHVAC_WaterToAirHeatPumpFields::AirOutletNodeName, false).get());
+  EXPECT_EQ("OutdoorAir:Mixer", idf_hp.getString(ZoneHVAC_WaterToAirHeatPumpFields::OutdoorAirMixerObjectType, false).get());
+  EXPECT_NE("", idf_hp.getString(ZoneHVAC_WaterToAirHeatPumpFields::OutdoorAirMixerName).get());
+  EXPECT_EQ("Autosize", idf_hp.getString(ZoneHVAC_WaterToAirHeatPumpFields::CoolingSupplyAirFlowRate, false).get());
```

Comment:
* I like the fact that you added a new test
* Maybe we should actually make it a full, targeted test. I see too much EXPECT_NE("", xxx) when I would like to see actual specific checks
* Don't use EXPECT_EQ / EXPECT_NE with an empty string. Use `EXPECT_TRUE(idf_hp.isEmpty(index));` instead